src/behavior.scm

```diff
@@ -758,38 +838,92 @@
+;(psi-rule (list (DefinedPredicate "Speech started?")) (DefinedSchemaNode "Speech started? action") speech-demand-satisfied (stv 1 1) speech-demand)
+;(psi-rule (list (DefinedPredicate "Speech ongoing?")) (DefinedSchemaNode "Speech ongoing? action") speech-demand-satisfied (stv 1 1) speech-demand)
+;(psi-rule (list (DefinedPredicate "Speech ended?")) (DefinedSchemaNode "Speech ended? action") speech-demand-satisfied (stv 1 1) speech-demand)
+;(psi-rule (list (DefinedPredicate "Skip Interaction?")) (DefinedSchemaNode "Keep alive") speech-demand-satisfied (stv 1 1) speech-demand)
+
+
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Interaction requested"))(DefinedSchemaNode "Interaction requested action") demand-satisfied (stv 1 1) face-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "New arrival sequence")) (DefinedSchemaNode "New arrival sequence action") demand-satisfied (stv 1 1) face-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Someone left")) (DefinedSchemaNode "Someone left action") demand-satisfied (stv 1 1) face-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Interact with people")) (DefinedSchemaNode "Interact with people action") demand-satisfied (stv 1 1) face-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Nothing is happening")) (DefinedSchemaNode "Nothing is happening action") demand-satisfied (stv 1 1) face-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Speech started?")) (DefinedSchemaNode "Speech started? action") speech-demand-satisfied (stv 1 1) speech-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Speech ongoing?")) (DefinedSchemaNode "Speech ongoing? action") speech-demand-satisfied (stv 1 1) speech-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Speech ended?")) (DefinedSchemaNode "Speech ended? action") speech-demand-satisfied (stv 1 1) speech-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Listening started?")) (DefinedSchemaNode "Listening started? action") speech-demand-satisfied (stv 1 1) speech-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Listening?")) (DefinedSchemaNode "Listening? action") speech-demand-satisfied (stv 1 1) speech-demand)
+(psi-rule (list (NotLink(DefinedPredicate "Skip Interaction?")) (DefinedPredicate "Listening ended?")) (DefinedSchemaNode "Listening ended? action") speech-demand-satisfied (stv 1 1) speech-demand)
+
+(psi-rule (list (DefinedPredicate "Skip Interaction?")) (DefinedSchemaNode "Keep alive") speech-demand-satisfied (stv 1 1) speech-demand)
 ;; ------------------------------------------------------------------
```

Comment:
Please place these in a different file so that the main loop below is not modified.